kartothek/io/eager.py

```diff
@@ -393,6 +387,18 @@ def commit_dataset(
         Input partition to be committed.
 
     """
+    if output_dataset_uuid is not None:
+        warnings.warn(
+            "The keyword `concat_partitions_on_primary_index` is deprecated and will be removed in the next major release.",
```

Comment:
wrong keyword `concat_partitions_on_primary_index`

Reply:
is this deprecated?

Reply:
The argument is not used anywhere. Made it clear in the warning statement also.

Reply:
The argument has no use and hence put the warning. Made the warning statement more clear